### ISSUE TITLE
Cannot request access token for client with CLIENT_SECRET_BASIC

### ISSUE THREAD

**Issue by jrmcdonald** [spring-projects-experimental/spring-authorization-server#346]
**Describe the bug**
The [ClientSecretBasicAuthenticationConverter#L89](https://github.com/spring-projects-experimental/spring-authorization-server/blob/77208f05d91210b156c2ce4475793c7d0a0f2c33/oauth2-authorization-server/src/main/java/org/springframework/security/oauth2/server/authorization/web/ClientSecretBasicAuthenticationConverter.java#L89) uses the deprecated ClientAuthenticationMethod.BASIC. This means that if you have a registered client with the ClientAuthenticationMethod.CLIENT_SECRET_BASIC, you cannot request an access token as the check in [OAuth2ClientAuthenticationProvider.java#L103](https://github.com/spring-projects-experimental/spring-authorization-server/blob/93d16d4419440be4b89e0ff92ec267bf26db1a02/oauth2-authorization-server/src/main/java/org/springframework/security/oauth2/server/authorization/authentication/OAuth2ClientAuthenticationProvider.java#L103) fails.

**To Reproduce**
1. Run the sample authorization server in this repository
2. Request an access token:

```
curl --location --request POST 'http://localhost:9000/oauth2/token' \
--header 'Content-Type: application/x-www-form-urlencoded' \
--header 'Authorization: Basic bWVzc2FnaW5nLWNsaWVudDpzZWNyZXQ=' \
--data-urlencode 'grant_type=client_credentials' \
--data-urlencode 'scope=message.read'
```

3. Observe successful response
4. Modify the registered client authentication method to ClientAuthenticationMethod.CLIENT_SECRET_BASIC
5. Repeat the above request
6. Observe 401 response

**Expected behavior**
Response should be 200 with a valid access token.

**Comment by sjohnr** [comment]
@jrmcdonald, thanks for the bug report! As you can see, a community member is already working on this.

**Comment by jrmcdonald** [comment]
Hi @sjohnr, thanks for that, do you know which milestone this is likely to be included in?

**Comment by jgrandja** [comment]
@jrmcdonald This is now merged and ready for `0.2.0`.
